ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -2101,6 +2102,41 @@ public boolean validateMaterializedViewsFromRegistry(List<Table> cachedMateriali
     }
   }
 
+  private Materialization getMaterializationInvalidationInfo(MaterializedViewMetadata metadata)
+      throws TException, HiveException {
+    MaterializationSnapshot mvSnapshot = MaterializationSnapshot.fromJson(metadata.creationMetadata.getValidTxnList());
+    if (mvSnapshot.getTableSnapshots() == null || mvSnapshot.getTableSnapshots().isEmpty()) {
```

Comment:
is this the only way to verify that's an ACID MV? could we inspect the MV create properties? we could be doing unnecessary JSON parse.

Reply:
This is not about verifying whether this is an ACID MV or a non native format which supports snapshots. The invalidation info depends on the source tables.
I refactored this to make the decision on the first source table storage format since mixed mode is not supported yet.

Reply:
👍